server/commands/notificationUpdater.ts

```diff
@@ -22,16 +23,16 @@ type Props = {
  */
 export default async function notificationUpdater({
   notification,
-  markAsViewed,
-  archive,
+  viewedAt,
+  archivedAt,
   ip,
   transaction,
 }: Props): Promise<Notification> {
-  if (markAsViewed) {
-    notification.viewedAt = new Date();
+  if (viewedAt) {
```

Comment:
I think this will need the same `isUndefined` logic to enable marking as unviewed?

Reply:
If we want to allow that, yes. I assumed it's not supposed to be reverted.

Reply:
Yes, it should be revertable as you can mark notifications as unread like an email inbox 👍

Reply:
Enabling the same in bulk update as well.